src/rebar_packages.erl

```diff
@@ -393,11 +393,37 @@ rm_ws(R) ->
     ec_semver:parse(R).
 
 valid_vsn(Vsn) ->
+    TVsn = fetch_transitive_vsn(Vsn),
     %% Regepx from https://github.com/sindresorhus/semver-regex/blob/master/index.js
     SemVerRegExp = "v?(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*))?"
         "(-[0-9a-z-]+(\\.[0-9a-z-]+)*)?(\\+[0-9a-z-]+(\\.[0-9a-z-]+)*)?",
-    SupportedVersions = "^(>=?|<=?|~>|==)?\\s*" ++ SemVerRegExp ++ "$",
-    re:run(Vsn, SupportedVersions, [unicode]) =/= nomatch.
+    SupportedVersions = "^" ++ vsn_pre_pattern() ++ SemVerRegExp ++ "$",
+    re:run(TVsn, SupportedVersions, [unicode]) =/= nomatch.
+
+vsn_pre_pattern() ->
+    "(>=?|<=?|~>|==)?\\s*".
+
+fetch_transitive_vsn(Vsn) ->
+    %% for removing `and` and `or` from dependecy and getting the highest version
+    case re:split(Vsn, <<"\s*or\s*|\s*and\s*">>) of
```

Comment:
I'm pretty sure that this allows `1.5.2-sorry` to be seen as versions `1.5.2-s` and `rry` since the specified space is made optional by the `*` pattern. Note also that it wouldn't handle other whitespace (like tabs) because `\s` is just a space (` `) and if you want the regex whitespace character class (`\s`) you need to double-escape it as `\\s`

Reply:
Do note that if someone writes a label condition such as `1.0.0-preor2.0.0` this now fetches version 2.0.0 and can be used to inject weird package versions for rebar3.